website/source/guides/identity/policy-templating.html.md

```diff
@@ -0,0 +1,607 @@
+
+1. [Write templated ACL policies](#step1)
+1. [Deploy your policy](#step2)
+1. [Setup an entity and a group](#step3)
+1. [Test the ACL templating](#step4)
+
+
+### <a name="step1"></a>Step 1: Write templated ACL policies
+
+Policy authors can pass in a policy path containing templating delimiters which
+is double curly braces (**`{{<parameter>}}`**).
+
+
+#### Available Templating Parameters
+
+|                                    Name                                |                                    Description                               |
+| :--------------------------------------------------------------------- | :--------------------------------------------------------------------------- |
+| `identity.entity.id`                                                   | The entity's ID                                                              |
+| `identity.entity.name`                                                 | The entity's name                                                            |
+| `identity.entity.metadata.<<metadata key>>`                            | Metadata associated with the entity for the given key                        |
+| `identity.entity.aliases.<<mount accessor>>.id`                        | Entity alias ID for the given mount                                          |
+| `identity.entity.aliases.<<mount accessor>>.name`                      | Entity alias name for the given mount                                        |
+| `identity.entity.aliases.<<mount accessor>>.metadata.<<metadata key>>` | Metadata associated with the alias for the given mount and metadata key      |
+| `identity.groups.ids.<<group id>>.name`                                | The group name for the given group ID                                        |
+| `identity.groups.names.<<group name>>.id`                              | The group ID for the given group name                                        |
+| `identity.groups.names.<<group id>>.metadata.<<metadata key>>`         | Metadata associated with the group for the given key                         |
+| `identity.groups.names.<<group name>>.metadata.<<metadata key>>`       | Metadata associated with the group for the given key                         |
+
+**Example:** The following policy a user to change their own password for those
+`userpass` users.
+
+```hcl
+path "auth/userpass/users/{{identity.entity.aliases.auth_userpass_6671d643.name}}/password" {
+  capabilities = [ "update" ]
+}
+```
+
+> **NOTE:** Identity groups are not directly attached to a token and an entity
+can be associated with multiple groups.  Therefore, in order to reference a
+group, the **group ID** or **group name** must be provided (e.g.
+`identity.groups.ids.59f001d5-dd49-6d63-51e4-357c1e7a4d44.name`).
+
+---
+
+#### Write the following policies:  
+
+**`user-tmpl.hcl`**
+
+```hcl
+# Grant permissions on user specific path
+path "user-kv/{{identity.entity.name}}/*" {
```

Comment:
I think it probably makes sense to have example use the KVv2 policy requirement.  This will eventually be the default implementation for kv.

Reply:
Updated the guide to use `kv-v2` instead (https://github.com/hashicorp/vault/pull/5226/commits/e9b2aaa27b2efdf86bf24911bf55dffe6d21194c).  Also, I found various typos in command, API payload example, etc. and they are fixed.